components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -1147,6 +1147,87 @@ void JsonRpcService::OnGetERC20TokenAllowance(
   std::move(callback).Run(args->at(0), mojom::ProviderError::kSuccess, "");
 }
 
+void JsonRpcService::GetERC20TokenBalances(
+    const std::vector<std::string>& token_contract_addresses,
+    const std::string& user_address,
+    const std::string& chain_id,
+    GetERC20TokenBalancesCallback callback) {
+  const auto& balance_scanner_contract_addresses =
+      GetEthBalanceScannerContractAddresses();
+  if (!base::Contains(balance_scanner_contract_addresses, chain_id)) {
+    std::move(callback).Run(
+        {}, mojom::ProviderError::kInvalidParams,
+        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+    return;
+  }
+  const auto& balance_scanner_contract_address =
+      balance_scanner_contract_addresses.at(chain_id);
+
+  if (token_contract_addresses.empty() || user_address.empty() ||
+      balance_scanner_contract_address.empty()) {
```

Comment:
I think we don't need to check for `balance_scanner_contract_address.empty()` as it comes from hardcoded map